user/tests/integration/wiring/no_fixture_i2c/i2c.cpp

```diff
@@ -0,0 +1,152 @@
+static const int WIRE1_ACQUIRE_BUFFER_SIZE = HAL_PLATFORM_I2C_BUFFER_SIZE(HAL_I2C_INTERFACE2) + 1;
+static const int TRACKER_MINIMUM_BUFFER_SIZE = 512;
+
+static hal_i2c_config_t allocateWireConfig(uint32_t bufferSize) {
+    hal_i2c_config_t config = {
+        .size = sizeof(hal_i2c_config_t),
+        .version = HAL_I2C_CONFIG_VERSION_2,
+        .rx_buffer = new (std::nothrow) uint8_t[bufferSize],
+        .rx_buffer_size = bufferSize,
+        .tx_buffer = new (std::nothrow) uint8_t[bufferSize],
+        .tx_buffer_size = bufferSize,
+        .flags = HAL_I2C_CONFIG_FLAG_FREEABLE
+    };
+    return config;
+}
+
+static void freeWireBuffers(hal_i2c_config_t * config){
+    free(config->tx_buffer);
+    free(config->rx_buffer);
+}
+
+hal_i2c_config_t acquireWireBuffer()
+{
+    return allocateWireConfig(WIRE_ACQUIRE_BUFFER_SIZE);
+}
+
+#if Wiring_Wire1
+hal_i2c_config_t acquireWire1Buffer()
+{
+    return allocateWireConfig(WIRE1_ACQUIRE_BUFFER_SIZE);
+}
+#endif
+
+test(I2C_00_hal_init_default_buffer)
+{
+    // Initializing the HAL with a null config is allowed and should allocate system defaults
+    assertEqual(hal_i2c_init(HAL_I2C_INTERFACE1, nullptr), (int)SYSTEM_ERROR_NONE);
+
+#if (PLATFORM_ID == PLATFORM_TRACKER) || (PLATFORM_ID == PLATFORM_TRACKERM)
+    // Tracker platforms should have buffers at least 512 bytes large, allocated by the system on startup
+    // IE trying to re-initialize the HAL with same size should fail
+    hal_i2c_config_t smallBuffers = allocateWireConfig(TRACKER_MINIMUM_BUFFER_SIZE);
+    hal_i2c_interface_t largeBufferInterface = (PLATFORM_ID == PLATFORM_TRACKER) ? HAL_I2C_INTERFACE2 : HAL_I2C_INTERFACE1;
+    assertEqual(hal_i2c_init(largeBufferInterface, &smallBuffers), (int)SYSTEM_ERROR_NOT_ENOUGH_DATA);
+    freeWireBuffers(&smallBuffers);
+#endif
+
+}
+
+test(I2C_01_test_acquire_wire_buffer) {
+    // Construct TwoWire object with user acquireWireBuffer() configuration structures, which should re-initialize the I2C HAL
+    Wire.begin();
+
+    // Attempt to re-initialize the HAL with different sized buffers to implicitly determine if the user buffers were used by the HAL
+    // System Defaults < User values from acquireWireBuffer < Values used here to re-initialize the HAL
+
+    // Equal size buffers should fail
+    hal_i2c_config_t wireSmallerBuffers = allocateWireConfig(WIRE_ACQUIRE_BUFFER_SIZE);
+    assertEqual(hal_i2c_init(HAL_I2C_INTERFACE1, &wireSmallerBuffers), (int)SYSTEM_ERROR_NOT_ENOUGH_DATA);
+    freeWireBuffers(&wireSmallerBuffers);
+
+    // Larger sized buffers should succeed
+    hal_i2c_config_t wireLargerBuffers = allocateWireConfig(WIRE_ACQUIRE_BUFFER_SIZE + 1);
+    assertEqual(hal_i2c_init(HAL_I2C_INTERFACE1, &wireLargerBuffers), (int)SYSTEM_ERROR_NONE);
+    freeWireBuffers(&wireLargerBuffers);
+
+    // Reinitializing with a NULL config is not allowed
+    assertEqual(hal_i2c_init(HAL_I2C_INTERFACE1, nullptr), (int)SYSTEM_ERROR_INVALID_ARGUMENT);
+
+#if Wiring_Wire1
```

Comment:
Nit: I'd probably move to a different test.

Reply:
ok, should I have a similar test for `acquireWire3Buffer()` just to be complete?

Reply:
Perhaps makes sense, yeah, for platforms with `Wiring_Wire3` which would as of right now be only Tracker.

Reply:
I can test `acquireWire3Buffer()` and `Wire3.begin()`, but then the `Wire1` tests fail with PMIC + RTC. 
I know Wire3 on tracker conflicts with UART pins and the `Wire` TWI handler, but it shouldn't affect `Wire1` right? IE `Wire1` and `Wire3` on tracker should be independently usable? 
I tried `Wire3.end()` before the RTC tests but they still fail. Not sure whats going wrong

Reply:
Yeah, sounds weird. Wire1 is TWIM1 and Wire3/Wire are TWIM0, so not sure what's going on.